dask-ml 2022.5.27

skupr-anaconda opened the pull request:
**Jira ticket:** [PKG-734](https://anaconda.atlassian.net/browse/PKG-734) (dask-ml-2022.5.27)

**The upstream data:**
Github releases:  https://github.com/dask/dask-ml/releases
[Diff between the latest and previous upstream releases](https://github.com/dask/dask-ml/compare/v1.9.0...v2022.5.27)
Requirements:
 * setup.py:  https://github.com/dask/dask-ml/blob/v2022.5.27/setup.py

**_Actions:_**

- Skip `py<36`

- Skip `linux-s390x` (`numba` isn't available)

- Add `wheel` to `host`

- Update `run` dependencies and pinnings

- Add `pip check`

- Fix `home` url with **https**

- Update `description`


**Additional info:**

<details>

**Package's statistics**

<details>

 * Priority B | effort: easy | Category: anaconda_affiliated | subcategory:  | pkg_type: python
 * Outdated platfroms: 'linux-64', 'linux-aarch64', 'linux-ppc64le', 'osx-64', 'osx-arm64', 'win-64'
 * Latest version: 2022.5.27
 * Release on PyPi:
    * version: 2022.5.27
    * date: 2022-05-27T14:17:47
* [PyPi history](https://pypi.org/project/dask-ml/#history)
 * Popularity: 
    * 3 months downloads: 30619.0
    * All time:  628217 downloads -  dask-ml  2022.5.27  Distributed and parallel machine learning using dask.  

</details>

**Other checks:**

<details>

1. - [x] https://github.com/dask/dask-ml/tree/v2022.5.27 exists
3. - [x] Check the pinnings
5. - [x] Additional research
    https://github.com/dask/dask-ml/issues
6. - [x] `dev_url`:
    https://github.com/dask/dask-ml
7. - [x] `doc_url`:
    https://ml.dask.org
8. - [x] Verify that the `build_number` is correct
9. - [x] Verify if the package needs `setuptools`
10. - [x] Verify if the package needs `wheel`
11. - [x] `pip` in test
12. - [x] Verify the test section
13. - [x] Verify if the package is `architecture specific`
14. - [x] Verify that private modules are not mentioned in the recipe For example: (_private_module)
15.  - [x] license_file: LICENSE.txt is present
16. - [x] license_family BSD is present
17. - [x] License: BSD-3-Clause
    - [x] License is `spdx` compliant
 * Check if the license identifier has correct name from the SPDX License List

| Identifier                           |
|:-------------------------------------|
| BSD-3-Clause                         |
| BSD-3-Clause-Attribution             |
| BSD-3-Clause-Clear                   |
| BSD-3-Clause-LBNL                    |
| BSD-3-Clause-Modification            |
| BSD-3-Clause-No-Military-License     |
| BSD-3-Clause-No-Nuclear-License      |
| BSD-3-Clause-No-Nuclear-License-2014 |
| BSD-3-Clause-No-Nuclear-Warranty     |
| BSD-3-Clause-Open-MPI                |
</details>

**Check dependency issues:**

<details>
../aggregate/dask-ml-feedstock/recipe/meta.yaml
Dependencies: ['pandas >=0.24.2', 'six', 'python >=3.6', 'scipy', 'setuptools_scm', 'setuptools', 'distributed >=2.4.0', 'scikit-learn >=0.23.0', 'multipledispatch >=0.4.9', 'packaging', 'dask-glm >=0.2.0', 'pip', 'dask >=2.4.0', 'numba >=0.51.0', 'numpy >=1.17.3']


noarch:
- py3.7:  No issues found
- py3.8:  No issues found
- py3.9:  No issues found
- py3.10:  No issues found

linux-64:
- py3.7:  No issues found
- py3.8:  No issues found
- py3.9:  No issues found
- py3.10:  Encountered problems while solving:
  - nothing provides __glibc >=2.17 needed by libgcc-ng-11.2.0-h1234567_0


linux-ppc64le:
- py3.7:  No issues found
- py3.8:  No issues found
- py3.9:  No issues found
- py3.10:  No issues found

linux-s390x:
- py3.7:  The following packages are missing from the target environment:
  - numba >=0.51.0

- py3.8:  The following packages are missing from the target environment:
  - numba >=0.51.0

- py3.9:  The following packages are missing from the target environment:
  - numba >=0.51.0

- py3.10:  The following packages are missing from the target environment:
  - numba >=0.51.0


osx-64:
- py3.7:  No issues found
- py3.8:  No issues found
- py3.9:  No issues found
- py3.10:  No issues found

osx-arm64:
- py3.8:  No issues found
- py3.9:  No issues found
- py3.10:  No issues found

win-64:
- py3.7:  No issues found
- py3.8:  No issues found
- py3.9:  No issues found
- py3.10:  No issues found

</details>

**Package Build Score: 18**


</details>



**Links:**
* [Anaconda Recipes feedstock](https://github.com/AnacondaRecipes/dask-ml-feedstock)
* [Update branch](https://github.com/AnacondaRecipes/dask-ml-feedstock/tree/2022.5.27)
* [conda-forge recipe](https://github.com/{upstream}/{feedstock_name}-feedstock)
* [PyPI](https://pypi.org/project/dask-ml)
* [Diff between upstream and feature branch](https://github.com/conda-forge/dask-ml-feedstock/compare/main...AnacondaRecipes:2022.5.27)
* [Diff between origin and feature branch](https://github.com/AnacondaRecipes/dask-ml-feedstock/compare/master...AnacondaRecipes:2022.5.27)
* [The last merged PRs](https://github.com/AnacondaRecipes/dask-ml-feedstock/pulls?q=is%3Apr+is%3Amerged+sort%3Aupdated-desc+)

**Updating the recipe:**
If the recipe needs additional modification the update branch can be modified. Note that the PR diffs are not updated with these changes.
```
git clone -b 2022.5.27 git@github.com:AnacondaRecipes/dask-ml-feedstock.git
```

skupr-anaconda committed:
DOR update to 2022.5.27

anaconda-pkg-build commented:
<details><summary>Linter check found the following problems:</summary>
The following problems have been found:

WARNING: /tmp/abs_faios7ubkn/clone/recipe/meta.yaml:0: uses_setuptools: The recipe uses setuptools in run depends
ERROR: /tmp/abs_faios7ubkn/clone/recipe/meta.yaml:1: missing_pip_check: For pypi packages, pip check should be present in the test commands
WARNING: /tmp/abs_faios7ubkn/clone/recipe/meta.yaml:17: avoid_noarch: noarch: python packages should be avoided
ERROR: /tmp/abs_faios7ubkn/clone/recipe/meta.yaml:24: missing_wheel: For pypi packages, wheel should be present in the host section
WARNING: /tmp/abs_faios7ubkn/clone/recipe/meta.yaml:62: http_url: http://github.com/dask/dask-ml is not https
WARNING: /tmp/abs_faios7ubkn/clone/recipe/meta.yaml:62: missing_license_url: The recipe is missing a license_url
WARNING: /tmp/abs_faios7ubkn/clone/recipe/meta.yaml:62: missing_doc_source_url: The recipe is missing a doc_source_url
Errors were found

</details>

skupr-anaconda commented:
An error on `s390x`:
```
conda.exceptions.ResolvePackageNotFound:
  - numba[version='>=0.51.0']
```
We can skip building on `s390x`

skupr-anaconda committed:
Update to 2022.5.7

- Skip py<36

- Skip linux-s390x (numba isn't available)

- Add wheel to host

- Update run dependencies and pinnings

- Add pip check

- Fix home url with https

- Update description

[PKG-734]

anaconda-pkg-build commented:
<details><summary>Linter check found the following problems:</summary>
The following problems have been found:

WARNING: /tmp/abs_204wv0abzd/clone/recipe/meta.yaml:68: missing_license_url: The recipe is missing a license_url
WARNING: /tmp/abs_204wv0abzd/clone/recipe/meta.yaml:68: missing_doc_source_url: The recipe is missing a doc_source_url
Warnings were found

</details>

varlackc reviewed (approved):
LGTM

anaconda-pkg-build commented:
<details><summary>Linter check found the following problems:</summary>
The following problems have been found:

WARNING: /tmp/abs_ebj7tgo_ej/clone/recipe/meta.yaml:68: missing_doc_source_url: The recipe is missing a doc_source_url
WARNING: /tmp/abs_ebj7tgo_ej/clone/recipe/meta.yaml:68: missing_license_url: The recipe is missing a license_url
Warnings were found

</details>

anaconda-pkg-build commented:
<details><summary>Linter check found the following problems:</summary>
The following problems have been found:

WARNING: /tmp/abs_f72mdcb4ys/clone/recipe/meta.yaml:68: missing_license_url: The recipe is missing a license_url
WARNING: /tmp/abs_f72mdcb4ys/clone/recipe/meta.yaml:68: missing_doc_source_url: The recipe is missing a doc_source_url
Warnings were found

</details>
